src/FakeAsyncClient.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Client\Utils;
+
+use Http\Client\HttpAsyncClient;
+use Http\Client\HttpClient;
+use Http\Client\Exception;
+use Http\Client\Utils\Promise\FulfilledPromise;
+use Http\Client\Utils\Promise\RejectedPromise;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Transform a HttpClient into a HttpAsyncClient
```

Comment:
should we rename the class to EmulateAsyncClient also ? (i like the emulate name, better than fake)